packages/pigeon/lib/dart_generator.dart

```diff
@@ -49,8 +49,58 @@ String _escapeForDartSingleQuotedString(String raw) {
       .replaceAll(r"'", r"\'");
 }
 
+String _calcCodecName(Api api) => '_${api.name}Codec';
+
+void _writeCodec(Indent indent, String codecName, Api api) {
+  indent.write('class $codecName extends StandardMessageCodec ');
+  indent.scoped('{', '}', () {
+    indent.writeln('const $codecName();');
+    if (getCodecClasses(api).isNotEmpty) {
+      indent.writeln('@override');
+      indent.write('void writeValue(WriteBuffer buffer, Object? value) ');
+      indent.scoped('{', '}', () {
+        bool first = true;
+        for (final EnumeratedClass customClass in getCodecClasses(api)) {
+          indent.write(
+              '${first ? '' : 'else '}if (value is ${customClass.name}) ');
```

Comment:
You could just output 'else ' at the end of the loop, instead of conditionally here and also just after the loop.

Reply:
The reason it is here is because we only want an else clause if `getCodecClasses(api).length > 0`.  This allows us to avoid invoking `getCodecClasses` twice.  Pulling out the else clause until after the loop would still require some variable to check that condition or calling the function twice.  It wouldn't be any cleaner imo.

Reply:
I don't follow. Your current code is:
```
        for (final EnumeratedClass customClass in getCodecClasses(api)) {
          indent.write(
              '${first ? '' : 'else '}if (value is ${customClass.name}) ');
          [...]
        }
        indent.write('else ');
```

Doing something at the beginning of every loop on every iteration after the first and also unconditionally after the loop identical to just doing something at the end of the loop. What extra variable would be required?

Reply:
Ahh, I remember what happened, sorry I was misremembering this.

Originally the code was written such that the `first` variable was used to decide if an else clause was needed.  Then I went back and just removed the codegen for the whole method if `getCodecClasses(api).length <= 0`.  After that it was logically correct so I didn't go back and shuffle it around.

The fact that you brought it up a few times leads me to believe you find it confusing, I'll refactor it.

Reply:
done